tests/jballerina-unit-test/src/test/resources/test-src/query/join-clause-negative.bal

```diff
@@ -325,15 +325,35 @@ function testOnClauseWithoutEquals() returns boolean {
 
     DeptPerson[] deptPersonList =
        from var person in personList
-       outer join Department dept in deptList
+       outer join var dept in deptList
        on person.id == dept.id
        select {
            fname : person.fname,
            lname : person.lname,
-           dept : dept.name
+           dept : dept?.name
        };
 }
 
 function condition(string name) returns boolean{
     return name == "Alex";
 }
+
+function testOuterJoinWithoutVar() returns boolean {
+    Person p1 = {id: 1, fname: "Alex", lname: "George"};
+    Person p2 = {id: 2, fname: "Ranjan", lname: "Fonseka"};
+
+    Department d1 = {id: 1, name:"HR"};
+    Department d2 = {id: 2, name:"Operations"};
+
+    Person[] personList = [p1, p2];
+    Department[] deptList = [d1, d2];
+
+    DeptPerson[] deptPersonList =
+       from Person person in personList
+       outer join Department dept in deptList
+       select {
+           fname : person.fname,
+           lname : person.lname,
+           dept : dept?.name
```

Comment:
Add negative tests for using `var` in type binding in join clause, but not considering nilable expression during access at `on`, `select` and other clauses.